### PR TITLE
Advanced search and new PokemonCard fields

### LINKED ISSUE #30: Advanced search parameters
We need an interface for searching with advanced search parameters. The implementation should need some changes in cardsearchbarview, as well as changes to the pokemon card class (to include more data) and the test card data object (to facilitate the filtering of results). I think the best way to do it UI wise is to make a popup menu like the save menu in deckbuilderview, with text fields to specify the search params. 

As for the actual filter parameters, we could filter by element which is already a part of the pokemon card class, maybe search by set, by whether it's a ex pokemon, etc.

### PR DESCRIPTION
**Pull request by Blackshadow8910**
Adds advanced search support and some new fields in the pokemon card class. We can display those new fields in the card viewer description pane if needed.

### CODE REVIEW

**Comment by Blackshadow8910** [comment]
I am also enabling the actual pokemon tcg api here, as it should be completely functional now.

File: src/main/java/data_access/image/HttpImageFetchWorker.java [hunk 1]
```diff
@@ -44,5 +52,21 @@ protected void done() {
             e.printStackTrace();
         }
     }
-    
+
+    private static BufferedImage readPlaceholderImage() {
+        try {
```
Comment (FredericoHideo) [inline src/main/java/data_access/image/HttpImageFetchWorker.java:57]: nice implementation!

File: src/main/java/data_access/pokemon/ArrayListCardDataAccessObject.java [hunk 2]
```diff
@@ -39,13 +32,61 @@ public Card getCard(String id) {
...
+            if (matchesFilter(card, filter)) {
                 results.add(card);
             }
         }
         return results;
     }
 
+    private boolean matchesFilter(Card card, PokemonGuruCardSearchFilter filter) {
+        if (!(card instanceof PokemonCard)) {
+            return false;
```
Comment (FredericoHideo) [inline src/main/java/data_access/pokemon/ArrayListCardDataAccessObject.java:44]: good

File: src/main/java/entity/PokemonGuruCardSearchFilter.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
+package entity;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class PokemonGuruCardSearchFilter {
```
Comment (FredericoHideo) [inline src/main/java/entity/PokemonGuruCardSearchFilter.java:8]: well designed filter

**Review by FredericoHideo — APPROVED** [review]

**Pull request merged**